umbral/umbral.py

```diff
@@ -129,14 +132,20 @@ def __bytes__(self):
         return self.to_bytes()
 
     def __eq__(self, other):
+        """
+        If both Capsules are activated, we compare only the activated components.
+        Otherwise, we compare only original components.
+        Done in constant time.
+        """
         if all(self.activated_components() + other.activated_components()):
-            activated_match = self.activated_components() == other.activated_components()
-            return activated_match
+            our_bytes = bytes().join(c.to_bytes() for c in  self.activated_components())
+            other_bytes = bytes().join(c.to_bytes() for c in  other.activated_components())
         elif all(self.original_components() + other.original_components()):
-            original_match = self.original_components() == other.original_components()
-            return original_match
+            our_bytes = bytes().join(c.to_bytes() for c in self.original_components())
+            other_bytes = bytes().join(c.to_bytes() for c in other.original_components())
         else:
             return False
+        return hmac.compare_digest(our_bytes, other_bytes)
```

Comment:
Why using hmac if it is already provided by cryptography.io?
https://cryptography.io/en/latest/hazmat/primitives/constant-time/#cryptography.hazmat.primitives.constant_time.bytes_eq

Reply:
Yes, we should definitely use the Cryptography.io one as it is also, probably, based on OpenSSL.

Reply:
I think, the reason is because `hmac` is a standard library. But if cryptography.io is better on this front - that's fine

Reply:
Yeah, I think we can go either way - happy to change to the cryptography.io version.  I just thought a sketch with the built-in constant time might be interesting, since we're able to compare bytes.

However, another question dawns on me this morning: since we were previously just comparing lists of elements which are compared in constant time anyway, do we even want to make this change?

It will only ever make a difference if somebody does something truly awful, like using their own `Point` or `BigNum` `class` in `Capsule`.